### ISSUE TITLE
make a real log stream instead of using print

### ISSUE THREAD

**Issue by cohenaj194** [ff14-advanced-market-search/AzerothAuctionAssassin#38]
GSA had a really pretty log stream, we should put actual effort into logging instead of using print statements everywhere.  I also need this written to a file so we can have a log when the app crashes.  If the GUI hits an exception during the run of mega_alerts.py then the whole thing goes down without much warning.  I need a log so i can debug issues. 

![image](https://github.com/ff14-advanced-market-search/AzerothAuctionAssassin/assets/17516896/73e77e98-d42f-4505-8ddf-b8c07cfb3dd8)

Need to update:
- https://github.com/ff14-advanced-market-search/AzerothAuctionAssassin/blob/main/mega_alerts.py
- basically everything in utils https://github.com/ff14-advanced-market-search/AzerothAuctionAssassin/tree/main/utils
- the gui app should also have some logging, right now it does nothing https://github.com/ff14-advanced-market-search/AzerothAuctionAssassin/blob/main/AzerothAuctionAssassin.py
